### DISCUSSION TITLE
Could not find Visual Studio VsDevCmd

### DISCUSSION THREAD

**Discussion by QuetzalcoutlDev** [HaxeFlixel/flixel#2843] [Q&A] [answered]
Download and install Visual Studio 2022 Community and various components, but when I want to try to build for Windows I get the following error:

  Warning: Could not find Visual Studio VsDevCmd
  Missing HXCPP_VARS
  Error: Could not automatically setup MSVC

**Comment by moxie-coder** [comment, marked as answer]
you need to use Visual Studio 2019, not 2022. this is coming from the backend of an issue with hxcpp tho, not flixel, and that it can’t find the compiler(s) it needs

**Reply by QuetzalcoutlDev** [reply]
Thanks for the help, I thought it was that I had downloaded something wrong, but something curious is that the Lime page does not specify which version to download and I ended up downloading VS 2022

**Reply by ACrazyTown** [reply]
I have 2022 and it works fine. This usually means people haven't installed the individual components such as the Windows SDK and MSVC @QuetzalcoutlDev

**Reply by QuetzalcoutlDev** [reply]
@ACrazyTown I installed the individual components, but it still wouldn't let me compile

**Reply by chrisdoesstuff69** [reply]
HOW DO YOU INSTALL VISUAL STUDIO 2019 I SWEAR I SEE IT ALL THE TIME BUT I CANT DOWNLOAD IT

**Comment by ACrazyTown** [comment]
> you need to use Visual Studio 2019, not 2022. this is coming from the backend of an issue with hxcpp tho, not flixel, and that it can’t find the compiler(s) it needs



> Error: Could not automatically setup MSVC

I've had both VS2019 and 2022 and never had this issue. I think something's not configured right with yours

**Reply by QuetzalcoutlDev** [reply]
It may be that, but in order not to complicate things, I will download the 2019 version
